x/auth/ante.go

```diff
@@ -231,9 +232,9 @@ func processSig(
 	return acc, res
 }
 
-func consumeSimSigGas(gasmeter sdk.GasMeter, pubkey crypto.PubKey, sig StdSignature, params Params) {
+func consumeSimSigGas(gasmeter sdk.GasMeter, pubkey crypto.PubKey, sig sdk.Signature, params Params) {
 	simSig := StdSignature{PubKey: pubkey}
-	if len(sig.Signature) == 0 {
+	if len(sig.GetSignature()) == 0 {
```

Comment:
what about creating a new `IsEmpty()` method to the interface ?

Reply:
This is the only occurrence - plus, `IsEmpty()` might turn out to be ambiguous. What is the emptiness check being carried upon? Should `IsEmpty()` return true when signature is not present or only when both the sig and the key are nil?